### ISSUE TITLE
False positive of `Style/IdenticalConditionalBranches` when conditional mutates duplicate line

### ISSUE THREAD

**Issue by martinemde** [rubocop/rubocop#12249]
## Expected behavior

When the conditional mutates the variable used in the duplication, the conditional should not be considered a violation of cop `Style/IdenticalConditionalBranches`.

## Actual behavior

This is identified as a duplicate branch, but because the duplicated line's behavior is modified by the conditional, it should not be caught by the cop, similar to #9982 where the conditional checks `x == 0` before `x += 1`.

## Steps to reproduce the problem

Extracted from Bundler, this removes a leading `-` and performs a different action depending on if the `-` existed.

```ruby
if version.delete_prefix!("-")
  version = version.split("-", 2).unshift(name)
  versions_by_name[name].delete(version)
else
  version = version.split("-", 2).unshift(name)
  versions_by_name[name] << version
end
```

The unsafe auto-correct is as follows, which is wrong:

```ruby
version = version.split("-", 2).unshift(name)
if version.delete_prefix!("-")
  versions_by_name[name].delete(version)
else
  versions_by_name[name] << version
end
```

The version I ended up with that avoids this issue is [in this PR](https://github.com/rubygems/rubygems/pull/6971/files#diff-2ea1208c335d87771b1c27ca8d6aa285ebb4fed566d6bf409f5df7ccb41c1ef8R35-R37).

## RuboCop version

`1.56.3`
